torchgeo/datasets/stacapidataset.py

```diff
@@ -0,0 +1,321 @@
+            assets=self.bands,
+            bounds_latlon=bounds,
+            resolution=self.res,
+            epsg=self._crs.to_epsg(),
+        )
+
+        dest = aoi.compute(scheduler="single-threaded")
+        return dest
+
+    def _merge_rasters(self, raster_list: List[DataArray]) -> Tensor:
+        """Merge a list of rasters.
+
+        Args:
+            raster_list: list of xarrays
+
+        Returns:
+            Tensor of merged xarray data.
+        """
+        time_dim = raster_list[0].shape[0]
+
+        # rioxarray only supports merges for 2D and 3D arrays so merge per time_step
+        # and later stack to one tensor
+        merged_rasters = []
+        for t in range(time_dim):
+            rasters_at_t = [r[t, ...] for r in raster_list]
+            merged_rasters.append(merge_arrays(rasters_at_t))
+
+        data: "np.typing.NDArray[np.float_]" = np.stack(merged_rasters)
+        tensor: Tensor = torch.tensor(data)  # type: ignore[attr-defined]
+
+        return tensor
+
+    def plot(
+        self,
+        sample: Dict[str, Tensor],
+        show_titles: bool = True,
+        suptitle: Optional[str] = None,
+    ) -> plt.Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample returned by :meth:`RasterDataset.__getitem__`
+            show_titles: flag indicating whether to show titles above each panel
+            suptitle: optional string to use as a suptitle
+
+        Returns:
+            a matplotlib Figure with the rendered sample
+        """
+        image = sample["image"].permute(1, 2, 0)
+        image = torch.clip(image / 10000, 0, 1)  # type: ignore[attr-defined]
+
+        fig, ax = plt.subplots(1, 1, figsize=(4, 4))
+        ax.imshow(image)
+        ax.axis("off")
+
+        if show_titles:
+            ax.set_title("Image")
+
+        if suptitle is not None:
+            plt.suptitle(suptitle)
+
+        return fig
+
+
+if __name__ == "__main__":
+
+    area_of_interest = {
+        "type": "Polygon",
+        "coordinates": [
+            [
+                [-148.56536865234375, 60.80072385643073],
+                [-147.44338989257812, 60.80072385643073],
+                [-147.44338989257812, 61.18363894915102],
+                [-148.56536865234375, 61.18363894915102],
+                [-148.56536865234375, 60.80072385643073],
+            ]
+        ],
+    }
+
+    time_of_interest = "2019-06-01/2019-08-01"
+
+    collections = (["sentinel-2-l2a"],)
+    intersects = (area_of_interest,)
+    datetime = (time_of_interest,)
+    query = ({"eo:cloud_cover": {"lt": 10}},)
+
+    rgb_bands = ["B04", "B03", "B02"]
+    ds = STACAPIDataset(
+        "./data",
+        bands=rgb_bands,
+        collections=["sentinel-2-l2a"],
+        intersects=area_of_interest,
+        datetime=time_of_interest,
+        query={"eo:cloud_cover": {"lt": 10}},
+    )
+
+    minx = -148.46876
+    maxx = -148.31072
+    miny = 61.0491
+    maxy = 61.12567489536982
```

Comment:
Torchgeo's GeoSampler does the bounding box query in the Rtree spatial index's CRS, which would be a UTM projection in this case.

```suggestion
    minx = 420688.14962388354
    maxx = 429392.15007465985
    miny = 6769145.954634559
    maxy = 6777492.989499866
```